internal/app/trustd/internal/reg/reg.go

```diff
@@ -31,15 +34,20 @@ func (r *Registrator) Register(s *grpc.Server) {
 
 // Certificate implements the securityapi.SecurityServer interface.
 func (r *Registrator) Certificate(ctx context.Context, in *securityapi.CertificateRequest) (resp *securityapi.CertificateResponse, err error) {
+	osRoot, err := safe.StateGet[*secrets.OSRoot](ctx, r.Resources, resource.NewMetadata(secrets.NamespaceName, secrets.OSRootType, secrets.OSRootID, resource.VersionUndefined))
+	if err != nil {
+		return nil, err
+	}
+
 	// TODO: Verify that the request is coming from the IP addresss declared in
```

Comment:
a ticket for this maybe

Reply:
this was this same way for a looong time, so proper answer I guess is to change the join flow to be based on some form of node attestation, but nothing changed in this PR